### PR TITLE
cpu: unified cortex-m[034]_common implementations

### PR DESCRIPTION
**Pull request by haukepetersen**
and the cleanup continues...

EDIT
~~I did not touch all CPUs yet, as I want to wait for #2852 to be merged.~~
 #2852 was merged.

Tested with one CPU from each Cortex family, the STM32F0 (M0), SAMD21 (M0+), STM32F1 (M3) STM32F3 & STM32F4 (both M4F).

### CODE REVIEW

**Comment by kaspar030** [comment]
nice one!

File: cpu/cortex-m_common/include/core_cm0.h [hunk 1]
```diff
@@ -138,8 +138,8 @@
     #error "Compiler generates FPU instructions for a device without an FPU (check __FPU_PRESENT)"
   #endif
 
-#elif defined ( __CSMC__ )      /* Cosmic */
-  #if ( __CSMC__ & 0x400)       /* FPU present for parser */
+#elif defined ( __CSMC__ )		/* Cosmic */
+  #if ( __CSMC__ & 0x400)		// FPU present for parser
```
Comment (gebart) [inline cpu/cortex-m_common/include/core_cm0.h]: Delete this change, you can reapply parts of #3016 to save on manual work.

File: cpu/cortex-m_common/include/core_cm0plus.h [hunk 2]
```diff
@@ -138,8 +138,8 @@
     #error "Compiler generates FPU instructions for a device without an FPU (check __FPU_PRESENT)"
   #endif
 
-#elif defined ( __CSMC__ )      /* Cosmic */
-  #if ( __CSMC__ & 0x400)       /* FPU present for parser */
+#elif defined ( __CSMC__ )		/* Cosmic */
+  #if ( __CSMC__ & 0x400)		// FPU present for parser
```
Comment (gebart) [inline cpu/cortex-m_common/include/core_cm0plus.h]: ditto

File: cpu/cortex-m_common/include/core_cm3.h [hunk 3]
```diff
@@ -138,8 +138,8 @@
     #error "Compiler generates FPU instructions for a device without an FPU (check __FPU_PRESENT)"
   #endif
 
-#elif defined ( __CSMC__ )      /* Cosmic */
-  #if ( __CSMC__ & 0x400)       /* FPU present for parser */
+#elif defined ( __CSMC__ )		/* Cosmic */
+  #if ( __CSMC__ & 0x400)		// FPU present for parser
```
Comment (gebart) [inline cpu/cortex-m_common/include/core_cm3.h]: ditto

File: cpu/cortex-m_common/include/core_cm4.h [hunk 4]
```diff
@@ -171,8 +171,8 @@
     #define __FPU_USED         0
   #endif
 
-#elif defined ( __CSMC__ )      /* Cosmic */
-  #if ( __CSMC__ & 0x400)       /* FPU present for parser */
+#elif defined ( __CSMC__ )		/* Cosmic */
+  #if ( __CSMC__ & 0x400)		// FPU present for parser
```
Comment (gebart) [inline cpu/cortex-m_common/include/core_cm4.h]: ditto

**Comment by gebart** [comment]
I think this is a step in the right direction for better maintainability.

**Comment by haukepetersen** [comment]
fixed tab issues

**Comment by haukepetersen** [comment]
here we go. This PR should now work for every cortex-based board.

**Comment by haukepetersen** [comment]
rebased on #3076 and adjusted the `saml21-xpro`

**Comment by haukepetersen** [comment]
rebased again.

**Comment by haukepetersen** [comment]
Could someone test and ack this, @PeterKietzmann, @kaspar030, @gebart, @OlegHahm? Otherwise it's building up a lot of dependecies... Thx!

File: cpu/cortexm_common/include/cpu.h [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2014 Freie Universität Berlin
+ * Copyright (C) 2015 Freie Universität Berlin
```
Comment (gebart) [inline cpu/cortexm_common/include/cpu.h]: you could leave the 2014 in there, i.e. 2014-2015

Comment (OlegHahm) [inline cpu/cortexm_common/include/cpu.h]: Actually, I think you *MUST*.

**Comment by gebart** [comment]
Update all the copyrights to include the entire time span in years year-now, e.g. 2014-2015

**Comment by gebart** [comment]
https://github.com/haukepetersen/RIOT/commit/ba9d259d67c50c2bc7ff851298fae9a6447ff238 commit message sepllnig: doxygen

**Comment by gebart** [comment]
I opened a PR at https://github.com/haukepetersen/RIOT/pull/8 for some minor improvements to the selection of toolchain prefixes. I plan on merging the Clang stuff from Mulle into cortex-m common later, but the PR is just a small preparation to make sure that all platforms follow the same pattern in the Makefile variables.

File: cpu/cortexm_common/include/cpu.h (hunk 5 above)
Comment (haukepetersen) [inline cpu/cortexm_common/include/cpu.h]: that was new to me. will fix it.

**Comment by haukepetersen** [comment]
fixed the copyrights, fixed the miss-spelled commit message and merged @gebarts fixes.

File: cpu/cortexm_common/include/cpu.h (hunk 5 above)
Comment (OlegHahm) [inline cpu/cortexm_common/include/cpu.h]: Maybe I was wrong about the MUST, but it seems to be a good idea not to change, but to expand, see http://stackoverflow.com/a/2391555 .

**Comment by gebart** [comment]
ACK, please squash, go when Travis is happy.

**Comment by haukepetersen** [comment]
suqashed and rebased. Go Travis, go.

**Comment by gebart** [comment]
Needs rebase...

**Comment by haukepetersen** [comment]
again?! Dammit...

**Comment by haukepetersen** [comment]
rebased.

**Comment by gebart** [comment]
I'm running a compile test locally on my machine

**Comment by gebart** [comment]
compile tests went fine on my machine

**Comment by PeterKietzmann** [comment]
ping

**Comment by gebart** [comment]
ACK holds, & go

**Comment by jfischer-phytec-iot** [comment]
:fireworks:

**Pull request merged**